WordPress/src/main/java/org/wordpress/android/WordPress.java

```diff
@@ -812,44 +815,49 @@ private void updatePushNotificationTokenIfNotLimited() {
                 public void run() {
-                    AppLog.i(T.UTILS, "App goes to background");
-                    // We're in the Background
-                    mIsInBackground = true;
-                    String lastActivityString = AppPrefs.getLastActivityStr();
-                    ActivityId lastActivity = ActivityId.getActivityIdFromName(lastActivityString);
-                    Map<String, Object> properties = new HashMap<String, Object>();
-                    properties.put("last_visible_screen", lastActivity.toString());
-                    if (mApplicationOpenedDate != null) {
-                        Date now = new Date();
-                        properties.put("time_in_app", DateTimeUtils.secondsBetween(now, mApplicationOpenedDate));
-                        mApplicationOpenedDate = null;
-                    }
-                    AnalyticsTracker.track(AnalyticsTracker.Stat.APPLICATION_CLOSED, properties);
-                    AnalyticsTracker.endSession(false);
-                    // Programmatically unregistering the ConnectionChangeReceiver here for Android >= N as per this:
-                    // https://developer.android.com/reference/android/net/ConnectivityManager.html
-                    // "Apps targeting Android 7.0 (API level 24) and higher do not receive this broadcast if they
-                    // declare the broadcast receiver in their manifest. Apps will still receive broadcasts if they
-                    // register their BroadcastReceiver with Context.registerReceiver() and that context is still valid."
-                    unregisterReceiver(ConnectionChangeReceiver.getInstance());
+                    onAppGoesToBackground();
                 }
             };
 
             this.mActivityTransitionTimer.schedule(mActivityTransitionTimerTask,
                     MAX_ACTIVITY_TRANSITION_TIME_MS);
         }
 
+        private void onAppGoesToBackground() {
+            AppLog.i(T.UTILS, "App goes to background");
+            mIsInBackground = true;
+            String lastActivityString = AppPrefs.getLastActivityStr();
+            ActivityId lastActivity = ActivityId.getActivityIdFromName(lastActivityString);
+            Map<String, Object> properties = new HashMap<String, Object>();
+            properties.put("last_visible_screen", lastActivity.toString());
+            if (mApplicationOpenedDate != null) {
+                Date now = new Date();
+                properties.put("time_in_app", DateTimeUtils.secondsBetween(now, mApplicationOpenedDate));
+                mApplicationOpenedDate = null;
+            }
+            AnalyticsTracker.track(AnalyticsTracker.Stat.APPLICATION_CLOSED, properties);
+            AnalyticsTracker.endSession(false);
+            // Methods onAppComesFromBackground / onAppGoesToBackground are only workarounds to track when the
+            // app goes or comes to background, but there are not 100% reliable, and here we could try
```

Comment:
Nitpick: It might be better to change `there` with `they` here. Also, do you think the following would be better overall?
```
app goes to or comes from background, but they are not 100% reliable, so we should avoid unregistering the receiver twice
```